### ISSUE TITLE
t/01_basic.t started to fail (with newest Test2::Suite?)

### ISSUE THREAD

**Issue by eserte** [akiym/Test2-Tool-Condition#1]
The test suite started to fail:
```
# Failed test at t/01_basic.t line 41.
# +----------------+-----+----------------+----+----------------+--------+
# | PATH           | LNs | GOT            | OP | CHECK          | LNs    |
# +----------------+-----+----------------+----+----------------+--------+
# |                |     | ARRAY(0x18b9f2 |    | <ARRAY>        | 28, 34 |
# |                |     | 8)             |    |                |        |
# |                |     |                |    |                |        |
# | [0]            | 27  | Test2::Event:: |    | <EVENT: Ok>    | 29     |
# |                |     | Fail=HASH(0x21 |    |                |        |
# |                |     | d40a8)         |    |                |        |
# |                |     |                |    |                |        |
# | [0] <blessed>  |     | Test2::Event:: | eq | Test2::Event:: |        |
# |                |     | Fail           |    | Ok             |        |
# |                |     |                |    |                |        |
# | [0]{pass}      |     | <DOES NOT EXIS |    | 0              |        |
# |                |     | T>             |    |                |        |
# |                |     |                |    |                |        |
# | [1]            |     | <DOES NOT EXIS |    | <EVENT: Diag>  | 30     |
# |                |     | T>             |    |                |        |
# |                |     |                |    |                |        |
# | [2]            |     | <DOES NOT EXIS |    | <EVENT: Diag>  | 33     |
# |                |     | T>             |    |                |        |
# +----------------+-----+----------------+----+----------------+--------+

# Failed test at t/01_basic.t line 58.
# +----------------+-----+----------------+----+----------------+--------+
# | PATH           | LNs | GOT            | OP | CHECK          | LNs    |
# +----------------+-----+----------------+----+----------------+--------+
# |                |     | ARRAY(0x219f72 |    | <ARRAY>        | 45, 51 |
# |                |     | 0)             |    |                |        |
# |                |     |                |    |                |        |
# | [0]            | 44  | Test2::Event:: |    | <EVENT: Ok>    | 46     |
# |                |     | Fail=HASH(0x13 |    |                |        |
# |                |     | c1ed8)         |    |                |        |
# |                |     |                |    |                |        |
# | [0] <blessed>  |     | Test2::Event:: | eq | Test2::Event:: |        |
# |                |     | Fail           |    | Ok             |        |
# |                |     |                |    |                |        |
# | [0]{pass}      |     | <DOES NOT EXIS |    | 0              |        |
# |                |     | T>             |    |                |        |
# |                |     |                |    |                |        |
# | [1]            |     | <DOES NOT EXIS |    | <EVENT: Diag>  | 47     |
# |                |     | T>             |    |                |        |
# |                |     |                |    |                |        |
# | [2]            |     | <DOES NOT EXIS |    | <EVENT: Diag>  | 50     |
# |                |     | T>             |    |                |        |
# +----------------+-----+----------------+----+----------------+--------+
# Seeded srand with seed '20190514' from local date.
t/01_basic.t .... 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/13 subtests 
```

Statistical analysis suggests that the problem started with Test2::Suite 0.000121 (@exodist: FYI):
```
****************************************************************
Regression 'mod:Test2::Suite'
****************************************************************
Name           	       Theta	      StdErr	 T-stat
[0='const']    	      1.0000	      0.0000	294328213456791296.00
[1='eq_0.000083']	     -0.0000	      0.0000	  -1.97
[2='eq_0.000097']	      0.0000	      0.0000	   2.57
[3='eq_0.000100']	      0.0000	      0.0000	   2.57
[4='eq_0.000111']	      0.0000	      0.0000	   2.37
[5='eq_0.000114']	      0.0000	      0.0000	   2.55
[6='eq_0.000115']	      0.0000	      0.0000	   0.71
[7='eq_0.000116']	      0.0000	      0.0000	   0.00
[8='eq_0.000117']	     -0.0000	      0.0000	  -4.02
[9='eq_0.000119']	      0.0000	      0.0000	   3.93
[10='eq_0.000120']	     -0.0000	      0.0000	  -1.72
[11='eq_0.000121']	     -1.0000	      0.0000	-107473467880320080.00

R^2= 1.000, N= 111, K= 12
****************************************************************
```

**Comment by exodist** [comment]
I introduced a new 'Fail' event that combines the failed assertion and the diagnostics into a single event, some Test2::Suite tools were switched to use it. These tests will need to be updated to support the new events.

I am thinking about how to avoid these kinds of issues in the future, I am thinking a test-testing tool that does not pay attention to the event types, but rather just pays attention to their effect. But you should not wait for that before updating your tests.

Sorry for the inconvenience.

**Comment by eserte** [comment]
Now it's the other way around: it passes with Test2::Suite 0.000122 and fails with older versions. Maybe the prereq version has to be adjusted?

**Comment by akiym** [comment]
@eserte Thank you. I've released these fixes with 0.03.
